Custom Header to enable ingress gateway tracing

twunderlich-grapl opened the pull request:
<!-- Thank you for your contribution to Grapl! Please take some time
to fill out this short template to help us review and understand your
PR. -->

### Which issue does this PR correspond to?
https://app.zenhub.com/workspaces/grapl-6036cbd36bacff000ef314f2/issues/grapl-security/issue-tracker/862
<!-- Please provide a link to the GitHub Issue this pull request is
meant to address. -->

### What changes does this PR make to Grapl? Why?
This is part of a larger distributed tracing project.
This particular piece adds a custom header to calls to the backend. This custom header is required in order to include ingress gateways in the trace.
Per https://www.consul.io/docs/connect/distributed-tracing#considerations

 `Requests through [Ingress Gateways](https://www.consul.io/docs/connect/gateways/ingress-gateway) will not be traced unless the header x-client-trace-id: 1 is set (see [hashicorp/consul#6645](https://github.com/hashicorp/consul/issues/6645)).`
<!-- Please describe the functional impact of these changes and the
rationale behind them. This will help us understand your approach to
solving the problem. -->

### How were these changes tested?
1. Tested as part of a larger tracing branch, where traces include the ingress gateway.
2. Regression tests should ensure that this doesn't break any functionality
<!-- Please describe how these changes were tested. There should be
sufficient detail that reviewers can replicate your testing
procedure. If the procedure is a manual one that's OK, your
description will help us work with you to get it automated. -->

Commit:
Enable ingress gateways to be traced

When tracing is on, the only way to include ingress gateways is by
having this custom header. Once that's done, the gateway is included
in the trace

twunderlich-grapl commented:
@andrea-grapl I _think_ I've added the header to all calls to our backend, however please let me know if there's somewhere else that we need to add this header to.

andrea-grapl reviewed (approved):
Looks good!

andrea-grapl reviewed (commented):
@twunderlich-grapl  Not sure where we want tracing, but we also have headers in `engagement_view/src/serviceWorker.ts`, will leave up to you if we want to add `x-client-trace-id` there too.

Commit:
Enable ig traces for service worker requests

Instrument service worker requests so we can see what's going on when
there's a problem

Commit:
Missed formatting the file
